osu.Game/Screens/Menu/DailyChallengeButton.cs

```diff
@@ -131,29 +130,37 @@ protected override void Update()
             }
         }
 
-        private void dailyChallengeChanged(bool postNotification)
+        private long? lastNotifiedDailyChallengeRoomId;
+
+        private void dailyChallengeChanged(ValueChangedEvent<DailyChallengeInfo?> info)
         {
             UpdateState();
 
             scheduledCountdownUpdate?.Cancel();
             scheduledCountdownUpdate = null;
 
-            if (info.Value == null)
+            if (this.info.Value == null)
             {
                 Room = null;
                 cover.OnlineInfo = TooltipContent = null;
             }
             else
             {
-                var roomRequest = new GetRoomRequest(info.Value.Value.RoomID);
+                var roomRequest = new GetRoomRequest(this.info.Value.Value.RoomID);
 
                 roomRequest.Success += room =>
                 {
                     Room = room;
                     cover.OnlineInfo = TooltipContent = room.Playlist.FirstOrDefault()?.Beatmap.BeatmapSet as APIBeatmapSet;
 
-                    if (postNotification)
+                    // We only want to notify the user if a new challenge recently went live.
+                    if (room.StartDate.Value != null
+                        && Math.Abs((DateTimeOffset.Now - room.StartDate.Value!.Value).TotalSeconds) < 600
```

Comment:
Before I go further - this is 10 minutes rather than 30, is it not? Is that ok?

Reply:
make a call, i think earthier is fine depending on perspective